src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                    case "false":
+                        return "";
+                    // all options below define macro "_CPPUNWIND":
+                    case "Async":
+                        return "/EHa";
+                    case "Sync":
+                        return "/EHsc";
+                    case "SyncCThrow":
+                        return "/EHs";
+                }
+            }
+
+            internal static string ConvertBasicRuntimeChecks(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported BasicRuntimeChecks: " + value);
+                    case "Default":
+                        return "";
+                    // all options below define macro "__MSVC_RUNTIME_CHECKS":
+                    case "StackFrameRuntimeCheck":
+                        return "/RTCs";
+                    case "UninitializedLocalUsageCheck":
+                        return "/RTCu";
+                    case "EnableFastChecks":
+                        return "/RTC1";
+                }
+            }
+        }
+
+        private static void AddList(List<string> cmd, string[] values)
+        {
+            foreach (string value in values)
+            {
+                Add(cmd, value);
+            }
+        }
+
+        private static void AddList(List<string> cmd, string prefix, string[] values)
+        {
+            foreach (string value in values)
+            {
+                Add(cmd, prefix, value);
+            }
+        }
+
+        private static void Add(List<String> cmd, string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                cmd.Add(value);
+            }
+        }
+
+        private static void Add(List<string> cmd, string prefix, string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                cmd.Add(prefix);
+                cmd.Add(value);
+            }
+        }
+
+        internal class Capture
+        {
+            [JsonProperty("compiler")]
+            public string Compiler { get { return "msvc-cl"; } }
+
+            [JsonProperty("cwd")]
+            public string Cwd { get; set; }
+
+            [JsonProperty("executable")]
+            public string Executable { get; set; }
+
+            [JsonProperty("cmd")]
+            public List<string> Cmd { get; set; }
+
+            [JsonProperty("env")]
+            public List<string> Env { get; set; }
+
+            [JsonProperty("stdout")]
+            public string StdOut { get; set; }
+
+            [JsonProperty("stderr")]
+            public string StdErr { get; set; }
+
+            public string ToJson()
+            {
+                var sb = new StringBuilder();
+                using (var writer = new StringWriter(sb))
+                {
+                    using (var textWriter = new JsonTextWriter(writer))
```

Comment:
Might be worth also to check whether `JsonTextWriter` closes the stream it takes as parameter.